packages/fuels-accounts/src/provider.rs

```diff
@@ -205,6 +213,7 @@ impl Provider {
         )?;
 
         let (status, receipts) = self.submit_with_feedback(tx.clone()).await?;
+        let receipts = receipts.ok_or(error!(ProviderError, "Submitting transaction failed"))?;
```

Comment:
We should incorporate the reason why it failed. Failure without receipts probably only happens for the `SqueezedOut` variant, it has a `reason` field we can include.